cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -1642,6 +1642,55 @@ const char* convert_toUTF8(int64_t context, const char* value, int32_t value_len
   return value;
 }
 
+// Calculate the levenshtein distance between two string values
+FORCE_INLINE
+gdv_int32 levenshtein_utf8_utf8(int64_t context, const char* in1, int32_t in1_len,
+                                const char* in2, int32_t in2_len) {
+  if (in1_len < 0 || in2_len < 0) {
+    gdv_fn_context_set_error_msg(context, "String length must be greater than 0");
+    return 0;
+  }
+  int32_t a, b, c, aux;
+  int32_t len_dist_1 = in1_len + 1;
+  int32_t len_dist_2 = in2_len + 1;
+  // dist[i][j] represents the Levenstein distance between the strings
+  int** dist;
+  if ((dist = (int**)malloc((len_dist_1 + 1) * sizeof(int*))) == NULL) {
+    gdv_fn_context_set_error_msg(context, "Insufficient space to allocate buffer");
+    return 0;
+  }
+  for (int i = 0; i <= in1_len; i++) {
+    if ((dist[i] = (int*)malloc(len_dist_2 * sizeof(int*))) == NULL) {
+      gdv_fn_context_set_error_msg(context, "Insufficient space to allocate buffer");
+      return 0;
+    }
+  }
+  for (int32_t i = 0; i <= in1_len; i++) dist[i][0] = i;
+  for (int32_t j = 1; j <= in2_len; j++) dist[0][j] = j;
+
+  for (int32_t j = 0; j < in2_len; j++) {
+    for (int32_t i = 0; i < in1_len; i++) {
+      if (in1[i] == in2[j]) {
+        dist[i + 1][j + 1] = dist[i][j];
```

Comment:
shouldn't this also be min(dist[i][j], 1 + dist[i][j + 1], 1 + dist[i + 1][j]), when in1[i] = in2[j]?
and when in1[i] != in2[j]  it is a min (1 + dist[i][j], 1 + dist[i][j + 1], 1 + dist[i + 1][j]) as properly coded from 1676..1680... 

Atleast In Java it is coded that way..